apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/welcome-tour/tour-card.js

```diff
@@ -180,26 +208,24 @@ function CardOverlayControls( { onMinimize, onDismiss } ) {
 	);
 }
 
-function TourRating( { isGutenboarding } ) {
+function TourRating() {
 	let isDisabled = false;
-	const tourRating = useSelect( ( select ) =>
-		select( 'automattic/wpcom-welcome-guide' ).getTourRating()
-	);
-	const { setTourRating } = useDispatch( 'automattic/wpcom-welcome-guide' );
 
-	if ( ! isDisabled && tourRating ) {
+	const [ tempRating, setTempRating ] = useState();
+	const tourRating = config.options?.tourRating?.useTourRating?.( tempRating ) ?? tempRating;
```

Comment:
I think that we may need to introduce some memoization, otherwise `recordTracksEvent` inside `useTourRating` could be called more than one time for each rating

Reply:
Yes, you are right. Thanks for flagging this. Partly why I don't like the approach with a single "smart" hook. I will try and break it apart so it doesn't rely on caching either.